core/utils/poly-look/src/visualisations/charts/bar-charts/horizontalBarChart/horizontalBarChart.js

```diff
@@ -0,0 +1,329 @@
+      scaleContainer.scale.domain(relevantData);
+    }
+  }
+
+  _updateExistingBars(bars) {
+    bars
+      .transition()
+      .duration(750)
+      .attr("x", 0)
+      .attr("width", initializingBarHeight)
+      .attr("y", (d) => {
+        for (let scaleContainer of this._yScales) {
+          if (!scaleContainer.id)
+            scaleContainer.scale(d.title) +
+              scaleContainer.scale.bandwidth() / 4;
+          if (d.group === scaleContainer.id)
+            return (
+              scaleContainer.scale(d.title) +
+              (scaleContainer.scale.bandwidth() - this._barWidth) / 2
+            );
+        }
+      })
+      .attr("height", this._barWidth)
+      .attr("fill", this._barColor)
+      .attr("class", "bar")
+      .transition()
+      .duration(750)
+      .attr("x", 0)
+      .attr("width", (d) => this._xScale(d.value));
+  }
+
+  _addEnteringBars(barGroups) {
+    barGroups
+      .append("rect")
+      .attr("x", 0)
+      .attr("width", initializingBarHeight)
+      .attr("y", (d) => {
+        for (let scaleContainer of this._yScales) {
+          if (!scaleContainer.id)
+            return (
+              scaleContainer.scale(d.title) +
+              scaleContainer.scale.bandwidth() / 4
+            );
+          if (d.group === scaleContainer.id)
+            return (
+              scaleContainer.scale(d.title) +
+              (scaleContainer.scale.bandwidth() - this._barWidth) / 2
+            );
+        }
+      })
+      .attr("height", this._barWidth)
+      .attr("fill", this._barColor)
+      .attr("class", "bar")
+      .transition()
+      .duration(750)
+      .delay((_, i) => i * 50)
+      .attr("x", 0)
+      .attr("width", (d) => this._xScale(d.value));
+  }
+
+  _addEnteringBarLabels(barLabels) {
+    barLabels
+      .append("text")
+      .attr("y", (d) => {
+        for (let scaleContainer of this._yScales) {
+          if (!scaleContainer.id)
+            return scaleContainer.scale(d.title) + barLabelTopMargin;
+          if (d.group === scaleContainer.id)
+            return (
+              scaleContainer.scale(d.title) +
+              scaleContainer.scale.bandwidth() -
+              barTextBottomMargin
+            );
+        }
+      })
+      .attr("class", "bar-label")
+      .attr("x", 0)
+      .text((d) => d.title)
+      .attr("fill", "transparent")
+      .style("font-size", "15px")
+      .transition()
+      .delay(1000)
+      .duration(500)
+      .attr("fill", this._groups ? this._barValueColor : "black");
+  }
+
+  _addEnteringGroupLabels(groupLabels) {
+    groupLabels
+      .append("text")
+      .attr(
+        "y",
+        (d) =>
+          this._yScales
+            .find((scaleContainer) => scaleContainer.id === d.id)
+            .scale.range()[0]
+      )
+      .attr("class", "group-label")
+      .attr("x", 0)
+      .text((d) => d.translation)
+      .attr("fill", "transparent")
+      .style("font-size", "15px")
```

Comment:
This one and the previous one are almost exactly the same. Also, there's some repetition of literals that maybe should be constants.